processor/src/operations/field_ops.rs

```diff
@@ -162,6 +162,54 @@ impl Process {
         self.stack.copy_state(1);
         Ok(())
     }
+
+    /// Computes a single turn of exp accumulation for the given inputs. The stack is arranged
+    /// as follows (from the top):
+    /// - least significant bit of the exponent in the previous trace if there's a expacc call,
+    /// otherwise ZERO - 1 element
+    /// - value which added to the accumulator in the previous trace is there's an expacc call
+    /// just before the current call, 1 otherwise - 1 element
+    /// - exponent of base for this turn - 1 element
+    /// - accumulated power of base so far - 1 element
+    /// - number which needs to be shifted to the right - 1 element
+    ///
+    /// To perform the operation we do the following:
+    /// 1. Pops top three elements off the stack and calculate the least significant bit of the
+    /// number `b`.
+    /// 2. Use this bit to decide if the current `base` raise to the power exponent needs to be
+    /// included in the accumulator.
+    /// 3. Update exponent with its square and the number b with one right shift.
+    /// 4. Pushes the calcuted new values to the stack in the mentioned order.
+    pub(super) fn op_expacc(&mut self) -> Result<(), ExecutionError> {
+        let mut exp = self.stack.get(2);
+        let mut acc = self.stack.get(3);
+        let mut b = self.stack.get(4);
+
+        // least significant bit of the number b.
+        let bit = b.as_int() & 1;
+
+        // value which would be incorporated in the accumulator.
+        let value = Felt::new((exp.as_int() - 1) * bit + 1);
+
+        // current value of acc after including the value based on whether the bit is
+        // 1 or not.
+        acc *= value;
+
+        // number `b` shifted right by one bit.
+        b = Felt::new(b.as_int() >> 1);
+
+        // exponent updated with its square.
+        exp *= exp;
+
+        self.stack.set(0, Felt::new(bit));
+        self.stack.set(1, value);
```

Comment:
I'd probably put `value` into one of user op helper registers - unless there is a good reason to keep it on the stack. (i.e., does it simply constraints in some way?)

Reply:
Yes- I've moved it to the helper register.